src/wp-includes/js/media-grid.js

```diff
@@ -268,7 +268,7 @@ document.addEventListener( 'DOMContentLoaded', function() {
 
 	// Open modal
 	function openModalDialog( item ) {
-		var id = item.id.replace( 'media-', '' ),
+		var id = item.dataset.id,
```

Comment:
I think this line is creating a problem.

On this branch open a media attachment modal view from Grid View and note then how the link for the `Edit more details` link contains `.../wp-admin/post.php?post=undefined&action=edit`, there `undefined should be the attachment database ID. The absence of ID from here stops updates from happening.

Reverting this changes restores normal functionality.

Reply:
Whoa, you're right! I will revert this change and open a separate PR to add an appropriate `data-id` (because the info is already held in an object and it will make for greater consistency for when the grid is used elsewhere)